tools/function_finder/function_finder_psx.py

```diff
@@ -164,13 +172,34 @@ def find_wip(o):
         for i, o in enumerate(output):
             # keep the in-source results as definitive
             if results[i] != None:
-                o[5] = results[i]["link"]
-                o[6] = results[i]["percent"]
+                o[6] = results[i]["link"]
+                o[7] = results[i]["percent"]
 
-    if args.use_call_trees:
-        base_url = (
+    base_url = (
             "https://raw.githubusercontent.com/Xeeynamo/sotn-decomp/gh-duplicates"
         )
+
+    if args.match_dups:
+        try:
+            f = open(f"gh-duplicates/duplicates.txt", "r")
+            dups_text = f.read()
+        except:
+            print("Could not find duplicates file")
+            f = None
+
+        for i, o in enumerate(output):
+            full_match = rf'1.00.+{o[1]}.+{o[0]}';
+            partial_match = rf'0.\d\d.+{o[1]}.+{o[0]}';
+
+            #print(f"Checking full match on {full_match}")
+            if re.search(full_match, dups_text) is not None:
+                #print("Full match found!")
+                o[5] = f"[Full]({base_url}/duplicates.txt#:~:text={o[1]})"
```

Comment:
[DOM text fragments](https://developer.mozilla.org/en-US/docs/Web/URI/Fragment/Text_fragments) are pretty fresh and not widely supported but in theory this would let the browser scroll the duplicates list directly to a relevant text match (aka the function name). It works on Chromium browsers but FF seems kind of iffy at the moment